### PR TITLE
LKE-6922 feat(space): Add create space API

### PR DESCRIPTION
**Pull request by aminerhanemi**

### CODE REVIEW

File: src/api/spaces/index.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+} = LkErrorKey;
+
+export class SpacesAPI extends Request {
+
+  /**
+   * Create a new space.
+   */
+  public createSpace(this: Request<ISpace>, params: ICreateSpaceParams) {
+    return this.request({
+      errors: [UNAUTHORIZED, FORBIDDEN, DATA_SOURCE_UNAVAILABLE, INVALID_PARAMETER],
```
Comment (aminerhanemi) [inline src/api/spaces/index.ts:25]: Still need to check which errors are returned

**Review by gmasclet — CHANGES_REQUESTED** [review]
This looks good to me beside a few misc changes :ok_hand:

File: src/api/spaces/index.ts (hunk 1 above)
Comment (aminerhanemi) [reply src/api/spaces/index.ts:16]: In which case the server should return INVALID_PARAMETER?

File: src/api/spaces/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import {LkErrorKey} from '../../http/response';
```
[the remark this answers is not in the document]
Comment (aminerhanemi) [reply src/api/spaces/index.ts:1]: I don't think it is needed in each file. the mention in the readme is enough. Maybe need to check with @enys .
I can add still add it

**Review by gmasclet — APPROVED** [review]

**Pull request merged**